src/hooks/useActionCenterSubscriber/useLimitOrderActionSubscriber.tsx

```diff
@@ -0,0 +1,394 @@
+  const quoteId = useAppSelector(selectActiveQuoteId)
+  const { currentData: ordersResponse } = useLimitOrdersQuery()
+  const toast = useToast({
+    render: ({ title, status, description, onClose }) => {
+      const handleClick = () => {
+        onClose()
+        onDrawerOpen()
+      }
+
+      const toastSx = {
+        '&:hover': {
+          cursor: 'pointer',
+          background: status === 'success' ? 'green.300' : 'red.300',
+        },
+      }
+
+      return (
+        <Flex
+          // We can't memo this because onClose prop comes from the render props
+          // eslint-disable-next-line react-memo/require-usememo
+          onClick={handleClick}
+          background={status === 'success' ? 'green.500' : 'red.500'}
+          color='white'
+          px={4}
+          py={2}
+          borderRadius='md'
+          // eslint-disable-next-line react-memo/require-usememo
+          sx={toastSx}
+        >
+          <Box py={1} me={2}>
+            {status === 'success' ? (
+              <Icon color={toastColor} as={TbCircleCheckFilled} height='20px' width='20px' />
+            ) : (
+              <Icon color={toastColor} as={TbCircleXFilled} height='20px' width='20px' />
+            )}
+          </Box>
+          <Box>
+            <CText fontWeight='bold' color={toastColor}>
+              {title}
+            </CText>
+            <CText color={toastColor}>{description}</CText>
+          </Box>
+        </Flex>
+      )
+    },
+  })
+  const openLimitOrders = useAppSelector(selectOpenLimitOrderActionsFilteredByWallet)
+  const quoteExpirationTimestamp = useAppSelector(selectActiveQuoteExpirationTimestamp)
+
+  const orderSubmissionMetadataFilter = useMemo(() => {
+    return { cowSwapQuoteId: quoteId ?? 0 }
+  }, [quoteId])
+
+  const limitOrderSubmissionMetadata = useSelectorWithArgs(
+    selectLimitOrderSubmissionMetadata,
+    orderSubmissionMetadataFilter,
+  )
+
+  const previousLimitOrderState = usePrevious(limitOrderSubmissionMetadata?.state)
+  const previousLimitOrderId = usePrevious(limitOrderSubmissionMetadata?.limitOrder.orderId)
+  const limitPrice = useAppSelector(selectActiveQuoteLimitPrice)
+
+  // Create action after user confirmed the intent of placing a limit order
+  useEffect(() => {
+    if (!limitOrderSubmissionMetadata) return
+    if (!sellAsset) return
+    if (!buyAsset) return
+    if (!quoteId) return
+    if (!activeQuote) return
+
+    const accountId = activeQuote.params.accountId
+
+    if (
+      limitOrderSubmissionMetadata.state ===
+        LimitOrderSubmissionState.AwaitingLimitOrderSubmission &&
+      previousLimitOrderState !== LimitOrderSubmissionState.AwaitingLimitOrderSubmission
+    ) {
+      const action = selectLimitOrderActionByCowSwapQuoteId(store.getState(), {
+        cowSwapQuoteId: quoteId,
+      })
+
+      if (action) return
+
+      dispatch(
+        actionSlice.actions.upsertAction({
+          id: uuidv4(),
```

Comment:
q: don't we want to use the `orderId` as an id here?

Reply:
Or perhaps `order.order.uid`

Reply:
I mean, I would not prefer to, action id is supposed to be unique, while maybe at some point for some reasons we could have multiple actions by order id if we change our pattern

Reply:
> > action id is supposed to be unique
> 
> Wouldn't _update_ with a new uuid void this and either create dupes as they would have different IDs? Whereas `uid` should indeed be unique and stay the same forever for a given order

It would create dupe but it would be an insert, not an update, updating is requiring an actual action.id, if you look at every updates we are passing the existing action, otherwise it would insert

I'm kind of agnostic regarding this, one or the other doesn't change so much from my point of view, anyway we are always updating by passing the whole object as everything is spread at update time

Reply:
yep but we are not adding a new uuid there right now, if we do this is a bug introduction rather than the actual implementation mistake, anyway, happy to update it later if we find like it sucks 🫶